### PR TITLE
fixed some clippy warnings

### PR DESCRIPTION
**Pull request by llogiq**

### CODE REVIEW

File: src/select_all.rs [hunk 1]
```diff
@@ -72,14 +72,14 @@ impl<A> Future for SelectAll<A>
     }
 
     fn schedule(&mut self, task: &mut Task) {
-        for f in self.inner.iter_mut() {
+        for f in &mut self.inner {
```
Comment (alexcrichton) [inline src/select_all.rs]: I personally prefer `.iter_mut()` instead of `&mut foo`, could this stay the same?

**Comment by llogiq** [comment]
Sure, can revert those once I had some sleep.

**Comment by alexcrichton** [comment]
Thanks!

**Comment by llogiq** [comment]
You're welcome. If I may ask: What's your rationale for preferring the explicit call instead of using `IntoIterator`?

**Comment by alexcrichton** [comment]
I personally find `for foo in bar.iter()` much easier to read than `for foo in &bar`. If you're new to Rust then the first is pretty clear what's happening where the latter just has this sigil `&` that seems odd.

**Pull request merged**